src/intelligence_layer/connectors/studio/studio.py

```diff
@@ -1,25 +1,57 @@
+import json
 import os
 from collections import defaultdict
-from collections.abc import Sequence
-from typing import Optional
+from collections.abc import Iterable, Sequence
+from typing import Generic, Optional, TypeVar
 from urllib.parse import urljoin
+from uuid import uuid4
 
 import requests
-from pydantic import BaseModel
+from pydantic import BaseModel, Field
 from requests.exceptions import ConnectionError, MissingSchema
 
+from intelligence_layer.connectors.base.json_serializable import (
+    SerializableDict,
+)
 from intelligence_layer.core.tracer.tracer import (  # Import to be fixed with PHS-731
     ExportedSpan,
     ExportedSpanList,
+    PydanticSerializable,
     Tracer,
 )
 
+Input = TypeVar("Input", bound=PydanticSerializable)
+ExpectedOutput = TypeVar("ExpectedOutput", bound=PydanticSerializable)
+
 
 class StudioProject(BaseModel):
     name: str
     description: Optional[str]
 
 
+class StudioExample(BaseModel, Generic[Input, ExpectedOutput]):
+    input: Input
+    expected_output: ExpectedOutput
+    id: str = Field(default_factory=lambda: str(uuid4()))
+    metadata: Optional[SerializableDict] = None
+
+
+class StudioDataset(BaseModel):
```

Comment:
We should probably put a comment here that this should mirror Dataset and Example. Also, StudioDataset has a docstring, but StudioExample doesn't

Reply:
Yes and no, at the moment it mirrors Dataset and Examples but, that's true, but that might not be the case in the future. But I will add the comment